### PR TITLE
[backend/frontend] Massive operations on global files and workbenches (#3016)

### LINKED ISSUE #7297: Important enhancement of the data / Import screen
## Use case

May be related to #3619.

Full rework of the Data / Import screen.

Especially 2 urgents things:

* Massive operation / retention policy on files uploaded here
* Massive operation / retention policy on workbenches

Work a little the layout?

### PR DESCRIPTION
**Pull request by Archidoit**
### Proposed changes
Rework screens of files and workbenches in Data/import
- Add tabs to manage files and workbenches
- Add paginations, filters, search and columns ordering
- Possibility to do massive operations on files and workbenches

Possibility to create background tasks on global files and workbenches
- new background task scope: IMPORT
- for capability KNOWLEDGE_KNASKIMPORT
- only operation authorized is DELETE

### Related issues
- MVP of https://github.com/OpenCTI-Platform/opencti/issues/3016
- https://github.com/OpenCTI-Platform/opencti/issues/7297

### CODE REVIEW

**Comment by Archidoit** [comment]
These filters are useless since the list doesn't change when we remove them  : 
image

Indeed pendingFiles always starts with import/pending, and global files are never linked to an entity

**Comment by Archidoit** [comment]
Filtering is not working 

image


+ added filters are not passing to the background task when doing a query task

**Comment by Archidoit** [comment]
After deleting a workbench in the Analyst workbenches tab, you need to reload the page to update the list (if no reload, the deleted workbench stays in the list with a grey bin icon)
image


same for files

**Comment by Archidoit** [comment]
In Analyst workbench tab, the workbenches are not clickable (we should be able to click on a line and be redirected to the workbench)

**Comment by Archidoit** [comment]
Not logical to display labels in this screen -> rather display markings 

image

**Comment by Archidoit** [comment]
Maybe add the possiblity to create a workbench in the Analyst workbench tab (like in the Import tab)

image

**Comment by Kedae** [comment]
Ready for new review

File: opencti-platform/opencti-front/src/private/components/data/DataTableToolBar.jsx [hunk 1]
```diff
@@ -648,6 +641,14 @@ class DataTableToolBar extends Component {
...
     );
 
+    const selectedElementsList = R.values(selectedElements);
+    const { selectedTypes } = this.getSelectedTypes();
+    let scope = 'KNOWLEDGE';
+    if (selectedElementsList.some(({ entity_type }) => entity_type === 'Vocabulary')) {
+      scope = 'SETTINGS';
+    } else if (selectedTypes.at(0) === 'InternalFile' && !this.state.scope) {
+      scope = 'IMPORT';
+    }
```
Comment (labo-flg) [inline opencti-platform/opencti-front/src/private/components/data/DataTableToolBar.jsx]: it's very bricolage... could we hide it under utility function `getBackgroundTaskScope(selectedList, selectedTypes)` and comment a bit the utility function ?

File: opencti-platform/opencti-front/src/private/components/data/import/ImportFilesContent.tsx [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+        message
+      }
+      creator {
+        name
+      }
+      entity_id
+      entity {
+        ... on AttackPattern {
+          name
+        }
```
Comment (labo-flg) [inline opencti-platform/opencti-front/src/private/components/data/import/ImportFilesContent.tsx]: would this be sufficient ? if only for display

```
... on StixObject {
          representative {
            main
          }
        }
```

File: opencti-platform/opencti-front/src/private/components/data/import/ImportWorkbenchesContent.tsx [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+        message
+      }
+      creator {
+        name
+      }
+      entity_id
+      entity {
+        ... on AttackPattern {
+          name
+        }
```
Comment (labo-flg) [inline opencti-platform/opencti-front/src/private/components/data/import/ImportWorkbenchesContent.tsx]: same remark

File: opencti-platform/opencti-graphql/src/domain/backgroundTask-common.js [hunk 4]
```diff
@@ -33,30 +35,32 @@ const areParentTypesKnowledge = (parentTypes) => parentTypes && parentTypes.flat
 // check a user has the right to create a list or a query background task
 export const checkActionValidity = async (context, user, input, scope, taskType) => {
   const { actions, filters: baseFilterObject, ids } = input;
-  const filters = (baseFilterObject && JSON.parse(baseFilterObject)?.filters) ?? [];
+  const filters = isFilterGroupNotEmpty(baseFilterObject)
+    ? (JSON.parse(baseFilterObject)?.filters ?? [])
+    : [];
```
Comment (labo-flg) [inline opencti-platform/opencti-graphql/src/domain/backgroundTask-common.js]: baseFilterObject seems to be a string, no ?
if so, isFilterGroupNotEmpty(baseFilterObject) is not relevant

File: opencti-platform/opencti-graphql/src/manager/taskManager.js [hunk 5]
```diff
@@ -150,34 +153,35 @@ const computeRuleTaskElements = async (context, user, task) => {
...
   // note that the query is filtered to allow only elements with matching confidence level
-  const data = await executeTaskQuery(context, user, task_filters, task_search, task_position);
+  const data = await executeTaskQuery(context, user, task_filters, task_search, scope, task_position);
   // const expectedNumber = data.pageInfo.globalCount;
   const elements = data.edges;
   // Apply the actions for each element
   for (let elementIndex = 0; elementIndex < elements.length; elementIndex += 1) {
     const element = elements[elementIndex];
-    if (!task_excluded_ids.includes(element.node.id)) {
+    if (!task_excluded_ids.includes(element.node.id)) { // keep only the elements that are not excluded (via uncoched checkboxes in UI)
```
Comment (labo-flg) [inline opencti-platform/opencti-graphql/src/manager/taskManager.js]:
```suggestion
    if (!task_excluded_ids.includes(element.node.id)) { // keep only the elements that are not excluded (via unticked checkboxes in UI)
```

File: opencti-platform/opencti-front/src/private/components/data/DataTableToolBar.jsx (hunk 1 above)
Comment (Kedae) [reply opencti-platform/opencti-front/src/private/components/data/DataTableToolBar.jsx]: Ok, I've explicited this extremely with a prop :)

File: opencti-platform/opencti-front/src/private/components/data/import/ImportFilesContent.tsx (hunk 2 above)
Comment (Kedae) [reply opencti-platform/opencti-front/src/private/components/data/import/ImportFilesContent.tsx]: Done

File: opencti-platform/opencti-front/src/private/components/data/import/ImportWorkbenchesContent.tsx (hunk 3 above)
Comment (Kedae) [reply opencti-platform/opencti-front/src/private/components/data/import/ImportWorkbenchesContent.tsx]: Done

**Comment by labo-flg** [comment]
Tested OK

**Review by labo-flg — APPROVED** [review]

**Pull request merged**